controllers/operatorconfig_controller.go

```diff
@@ -54,51 +71,341 @@ type OperatorConfigReconciler struct {
 func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
+	// Fetch the OperatorConfig operatorConfig
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, req.NamespacedName, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Add finalizer for this CR
+	if !controllerutil.ContainsFinalizer(operatorConfig, finalizer) {
+		controllerutil.AddFinalizer(operatorConfig, finalizer)
+		err := r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if operatorConfig.DeletionTimestamp != nil {
+		// clean up
+		err := r.CleanUpOpjects(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		controllerutil.RemoveFinalizer(operatorConfig, finalizer)
+		err = r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	filepaths := []string{"proxy.yaml", "manifests.yaml", "crane-ui-plugin.yaml"}
+
+	for _, f := range filepaths {
+		err := r.createResourcesFromFile(f, ctx, log, operatorConfig)
+		if err != nil {
+			log.Error(err, "Error creating resources")
+			return ctrl.Result{Requeue: true}, nil
+		}
+	}
+
+	log.Info("All needed resources are present")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorConfigReconciler) CleanUpOpjects(ctx context.Context) error {
+	filepaths := []string{"proxy.yaml", "crane-ui-plugin.yaml", "manifests.yaml"}
+
+	for _, f := range filepaths {
+		err := r.deleteResourcesFromFile(f, ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *OperatorConfigReconciler) createResourcesFromFile(path string, ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
 
-	err := createClusterTasks(ctx, log)
+	var decoder = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
 
+	data, err := getResources(path)
 	if err != nil {
-		log.Error(err, "Error creating cluster tasks")
-		return ctrl.Result{Requeue: true}, nil
-	} else {
-		log.Info("All the needed cluster task created")
+		return err
+	}
+
+	for _, resource := range data {
+		if len(resource) > 0 {
+			obj := unstructured.Unstructured{}
+			_, gvk, err := decoder.Decode([]byte(resource), nil, &obj)
+			if err != nil {
+				return err
+			}
+
+			switch gvk.Kind {
+			case "Deployment":
+				err = r.createDeployment(&obj, ctx, log, operatorConfig)
```

Comment:
rename the create methods `reconcileDeployment` instead create because you are essentially creating, checking and then updating

also wonder if creating a will help in cleaning the code:

```
reconcilersForGVK := map[string]func(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
   "Deployment": r.reconcileDeployment,
   "Service": r.reconcileService,
  "ConfigMap": r.reconcileConfigMap
}
    			_, gvk, err := decoder.Decode([]byte(resource), nil, &obj)
			if err != nil {
				return err
			}
			if fn, ok := reconcilersForGVK[gvk.Kind]; ok {
                             err := fn(&obj, ctx, log, operatorConfig)
                             if err != nil {
                                 return err
                             }
			}
		        return fmt.Errorf("unknown gvk or a better error message")
```